### ISSUE TITLE
Spark plugs fouling

### ISSUE THREAD

**Issue by SH-M** [Juanvvc/c172p-detailed#800]
Spark plugs can clog and eventually foul if the engine is allowed to idle too low for too long. Throttling up an engine with oil-soaked spark plugs can help clear them out.

The engine should be at or above 800 RPM. Failure to do so may cause spark plug fouling. If the plugs do foul (the engine will sound rough), running the engine at a higher RPM will blow them clear within a few seconds by doing so. 

Possible to simulate?

**Comment by gilbertohasnofb** [comment]
This actually sounds like a cool idea! I don't have in mind how to implement it, but I imagine it should be somehow possible. I know for a fact that the payware aircraft  A2A Cessna 172 for FSX does simulate this, so why couldn't we? :smile:

**Comment by legoboyvdlp** [comment]
The stoichiometric ratio is fifteen parts of air for each of fuel (15:1). At peak EGT you’re right there.
But when the engine is idling, let’s say below 1000RPMs, mixture full rich, that ratio is way lower, that is, there’s much more fuel than needed.

The lower the air fuel ratio, the colder the combustion and the higher the amount of fuel unburned. As you may know, a cold combustion rich of fuel leads to one thing: carbon.

Some of the carbon is ejected out of the exhaust pipe but a part of it deposits over the spark plugs, fouling them.
A fouled spark plug is unable to make the spark that starts the combustion, because the carbon deposits are isolating the two electrodes, thus preventing the spark to take place.

 An aircraft piston engine has two spark plugs for each cylinder, one connected to the left magneto, the other connected to the right magneto.

If the RPM drop over a magneto is higher that 125-150 RPM, then one or more spark plugs attached to that magneto are fouling


I think the AFR was exposed in the property tree in a recent jsbsim update. This might require C++ changes to the piston engine code to allow for fouling - not sure if this is currently possible.

**Comment by wlbragg** [comment]
>However, I think with a not-firing plug the ammount of fuel used should be the same.

Yes, it should still be drawing the same amount of fuel, its just not burning it.

**Comment by hbeni** [comment]
Already implemented in C182s, maybe you can port it easily:
https://github.com/HHS81/c182s/issues/439

**Comment by wlbragg** [comment]
Saw that, on my todo list, thanks!
